Implemented new virtual disk streaming I/O.

roliveri opened the pull request:
Implemented module-based disk IO streaming chain.

roliveri committed:
Implemented new virtual disk streaming I/O.

movitto commented:
@roliveri I went thought this and the code looks good. I didn't verify the DOS / MBR / GPT standards but successfully ran the spec suite (which are the testing the attached disks which have assumably been formatted with the corresponding data structures?) The was one small change I had to make to get the tests running locally, will comment inline (may just be due to an older dep on my system).

The FileIO class is intended to replace the current 'BlockFile' class correct? If so we can switch over the existing virtfs modules to use that instead and remove that from here. In any case, everything else looks good, have no problem w/ merging this

movitto commented on spec/spec_helper.rb:
```diff
@@ -0,0 +1,20 @@
+              when 'WARN'  then Logger::WARN
+              when 'INFO'  then Logger::INFO
+              when 'DEBUG' then Logger::DEBUG
+              else              Logger::OFF
```
I'm getting an "undefined constant Logger::OFF" here, changing it to one of the other values is a workaround

roliveri replied:
@movitto I converted this from Log4r to Logger, to eliminate the Log4r dependancy. I guess Logger doesn't have an "OFF" level. I'll change the default to "ERROR".

roliveri commented:
@movitto `FileIO` replaces `BlockFile` for regular files, which is what we need for most of the cases. We'll probably need a different module to access block special files, like `/dev/sda`.

roliveri committed:
Misc cleanup:

Added .travis.yml
Updated LICENSE.txt
Changed default log level in spec/spec_helper.rb

roliveri commented:
@movitto Now that this is merged, try converting some of your existing VirtFS FS modules to use this.
